src/core/symbology/qgssymbol.cpp

```diff
@@ -86,28 +86,31 @@ QPolygonF QgsSymbol::_getLineString( QgsRenderContext &context, const QgsCurve &
 
   QgsCoordinateTransform ct = context.coordinateTransform();
   const QgsMapToPixel &mtp = context.mapToPixel();
-  QPolygonF pts;
+  QgsLineString pts;
 
   //apply clipping for large lines to achieve a better rendering performance
   if ( clipToExtent && nPoints > 1 && !( context.flags() & QgsRenderContext::ApplyClipAfterReprojection ) )
   {
     const QgsRectangle e = context.extent();
     const double cw = e.width() / 10;
     const double ch = e.height() / 10;
-    const QgsRectangle clipRect( e.xMinimum() - cw, e.yMinimum() - ch, e.xMaximum() + cw, e.yMaximum() + ch );
-    pts = QgsClipper::clippedLine( curve, clipRect );
+    const QgsBox3d clipRect( QgsPoint( e.xMinimum() - cw, e.yMinimum() - ch ), QgsPoint( e.xMaximum() + cw, e.yMaximum() + ch ) ); // TODO also need to be clipped according to z axis
```

Comment:
it'll be faster to use the direct double xmin, double ymin etc constructor here, instead of creating the intermediate QgsPoint objects

Reply:
ok